### PR TITLE
add details about InstanceShutdownByProviderID

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Details what InstanceShutdownByProviderID should return if instance is deleted

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @cheftako 
/cc @andrewsykim

### CODE REVIEW

File: staging/src/k8s.io/cloud-provider/cloud.go [hunk 1]
```diff
@@ -186,6 +186,7 @@ type Instances interface {
 	// This method should still return true for instances that exist but are stopped/sleeping.
 	InstanceExistsByProviderID(ctx context.Context, providerID string) (bool, error)
 	// InstanceShutdownByProviderID returns true if the instance is shutdown in cloudprovider
+	// If the instance is deleted, this method should return false, or throw an error.
```
Comment (cheftako) [inline staging/src/k8s.io/cloud-provider/cloud.go:189]: I don't like 'or'. 
I think we should suggest the boolean behavior. 
Error should be an indicator that something went wrong and we could not determine the answer. (Eg. failed to connect to the relevant service or the function has not been implemented)
Also I believe if the instance has been deleted, then it has been shutdown and we should be returning true.

Comment (Sh4d1) [reply staging/src/k8s.io/cloud-provider/cloud.go:189]: @cheftako returning true completely ignores the delete in the controller, and the nodes is never deleted :sweat_smile: a `cloudprovider.InstanceNotFound` makes sense for me though (more than `false` anyway` :smile: )

Comment (Sh4d1) [reply staging/src/k8s.io/cloud-provider/cloud.go:189]: (Today returning false or an error have the same behaviour, except the error is logged)

**Comment by Sh4d1** [comment]
@cheftako could we merge this one and open an issue to better handle this function?

**Pull request closed without merging**